Composed/Core/DataSources/SingleItemDataStore.swift

```diff
@@ -0,0 +1,59 @@
+/**
+ A `DataSource` that contains a single item. It is backed by a `SingleItemDataStore`
+ */
+open class SingleElementDataSource<Element>: DataSource {
+
+    public weak var updateDelegate: DataSourceUpdateDelegate?
+    public private(set) var element: Element
+
+    public var numberOfSections: Int { return 1 }
+
+    public var indexPath: IndexPath? {
+        return isEmpty ? nil : IndexPath(item: 0, section: 0)
+    }
+
+    public var isEmpty: Bool {
+        return numberOfElements(in: 0) > 0
+    }
+
+    /**
+     Create a new `SingleItemDataSource` with the provided element
+
+     - parameter element: The element to populate the backing store with
+     */
+    public init(element: Element) {
+        self.element = element
+    }
+
+    public func numberOfElements(in section: Int) -> Int {
```

Comment:
I actually just came up with a scenario where it's useful to have this return something other than `0` and `1` here. I will explain via PR in other project, but noting here and I think it'd be useful for this to be `open`